File: frontend/src/services/dashboardService.js
// src/services/dashboardService.js

import api from "../api/axios";

const BASE_URL = "/dashboard";

export const DashboardService = {

  /* =========================================================
     SUMMARY
  ========================================================= */

  async getSummary(financial_year) {

    const response = await api.get(
      `${BASE_URL}/summary`,
      {
        params: {
          financial_year,
        },
      }
    );

    return response.data?.data;
  },

  /* =========================================================
     FINANCIAL ANALYTICS
  ========================================================= */

  async getFinancialAnalytics(financial_year) {

    const response = await api.get(
      `${BASE_URL}/financial-analytics`,
      {
        params: {
          financial_year,
        },
      }
    );

    return response.data?.data || [];
  },

  /* =========================================================
     RECENT INCOMES
  ========================================================= */

  async getRecentIncomes() {

    const response = await api.get(
      `${BASE_URL}/recent-incomes`
    );

    return response.data?.data || [];
  },

  /* =========================================================
     RECENT EXPENSES
  ========================================================= */

  async getRecentExpenses() {

    const response = await api.get(
      `${BASE_URL}/recent-expenses`
    );

    return response.data?.data || [];
  },

  /* =========================================================
     RECENT RENTALS
  ========================================================= */

  async getRecentRentals() {

    const response = await api.get(
      `${BASE_URL}/recent-rentals`
    );

    return response.data?.data || [];
  },

  /* =========================================================
     EXPENSE BREAKDOWN
  ========================================================= */

  async getExpenseBreakdown(financial_year) {

    const response = await api.get(
      `${BASE_URL}/expense-breakdown`,
      {
        params: {
          financial_year,
        },
      }
    );

    return response.data?.data || [];
  },

  /* =========================================================
     INCOME BREAKDOWN
  ========================================================= */

  async getIncomeBreakdown(financial_year) {

    const response = await api.get(
      `${BASE_URL}/income-breakdown`,
      {
        params: {
          financial_year,
        },
      }
    );

    return response.data?.data || [];
  },

  /* =========================================================
     FILTER OPTIONS
  ========================================================= */

  async getFilterOptions() {

    const response = await api.get(
      `${BASE_URL}/filter-options`
    );

    return response.data?.data || {
      financial_years: [],
      expense_categories: [],
    };
  },
};

File: backend/app/services/dashboard_service.py
# app/services/dashboard_service.py

from app.repositories.dashboard_repo import DashboardRepository


class DashboardService:

    def __init__(self):

        self.repo = DashboardRepository()

    async def get_dashboard_summary(
        self,
        user_id: str,
        financial_year: str
    ):

        data = await self.repo.get_dashboard_summary(
            user_id,
            financial_year
        )

        return {
            "success": True,
            "data": data
        }

    async def get_financial_analytics(
        self,
        user_id: str,
        financial_year: str
    ):

        data = await self.repo.get_financial_analytics(
            user_id,
            financial_year
        )

        return {
            "success": True,
            "data": data
        }

    async def get_recent_incomes(
        self,
        user_id: str
    ):

        data = await self.repo.get_recent_incomes(
            user_id
        )

        return {
            "success": True,
            "data": data
        }

    async def get_recent_expenses(
        self,
        user_id: str
    ):

        data = await self.repo.get_recent_expenses(
            user_id
        )

        return {
            "success": True,
            "data": data
        }

    async def get_recent_rentals(
        self,
        user_id: str
    ):

        data = await self.repo.get_recent_rentals(
            user_id
        )

        return {
            "success": True,
            "data": data
        }

    async def get_expense_breakdown(
        self,
        user_id: str,
        financial_year: str
    ):

        data = await self.repo.get_expense_breakdown(
            user_id,
            financial_year
        )

        return {
            "success": True,
            "data": data
        }

    async def get_income_breakdown(
        self,
        user_id: str,
        financial_year: str
    ):

        data = await self.repo.get_income_breakdown(
            user_id,
            financial_year
        )

        return {
            "success": True,
            "data": data
        }

    async def get_filter_options(
        self,
        user_id: str
    ):

        data = await self.repo.get_filter_options(
            user_id
        )

        return {
            "success": True,
            "data": data
        }